lib/nexthop_group.c

```diff
@@ -481,7 +481,7 @@ void nexthop_group_mark_duplicates(struct nexthop_group *nhg)
 		for (ALL_NEXTHOPS_PTR(nhg, prev)) {
 			if (prev == nexthop)
 				break;
-			if (nexthop_same_firsthop(nexthop, prev)) {
+			if (nexthop_same(nexthop, prev)) {
```

Comment:
hmm - I'm not sure the semantics of nexthop_same() are quite ... the same. nexthop_same() will return false if two nexthops have different "weight" values, for example, even if their gateway info is the same?

Reply:
yea it was only a subset.  what "shouldnt" be matched? its a weird function to just have sitting around to match dupes if you ask me. i figured it would be fine to leave labels/weights being compared.

i can add a "nexthop_cmp_common()" or something that this will just call as well as nexthop_same() so at least we are using the same code.

Reply:
Just trying to confirm that the change in behavior is really what's intended. If we saw a group with {nexthop A, weight 2, nexthop A weight 3} the existing code will identify one of those a duplicate, because it's testing the gateway. the new behavior would allow both to be included in the group, because they differ in weight and that would be considered "different" now.
Is it meaningful to have two nexthops with the same gateway, one with labels and one without, for example? it's hard to image that that's useful, but the change here would permit it.

Reply:
yea this function is so old (3yrs i believe) that im not sure what should be included. Yes it definitely changes, but I assumed that's what we wanted. We can bring it up tomorrow.

Reply:
I went through this a bit with Donald, and tried it out with iproute, and netlink (at least) really will accept the groups that this change will produce, so I'll go ahead and merge this. If some consumer of nhgs needs to be more strict, it'll have to implement its own filtering logic I guess.